src/test/java/org/junitpioneer/jupiter/EnvironmentVariableUtilsTests.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatCode;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.security.AccessControlException;
+import java.security.Policy;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+@DisplayName("EnvironmentVariableUtils")
+@ClearEnvironmentVariable(key = "TEST")
+class EnvironmentVariableUtilsTests {
+
+	@Test
+	@SetSystemProperty(key = "java.security.policy", value = "file:src/test/resources/default-testing.policy")
+	void shouldThrowAccessControlExceptionWithDefaultSecurityManager() {
+		executeWithSecurityManager(() -> assertThatThrownBy(() -> EnvironmentVariableUtils.set("TEST", "TEST"))
+				.isInstanceOf(AccessControlException.class));
+	}
+
+	@Test
+	@SetSystemProperty(key = "java.security.policy", value = "file:src/test/resources/reflect-permission-testing.policy")
+	void shouldModifyEnvironmentVariableIfPermissionIsGiven() {
+		executeWithSecurityManager(() -> {
+			assertThatCode(() -> EnvironmentVariableUtils.set("TEST", "TEST")).doesNotThrowAnyException();
+			assertThat(System.getenv("TEST")).isEqualTo("TEST");
+		});
+	}
+
+	/*
+	 * This needs to be done during the execution of the test method and cannot be moved to setup/tear down
+	 * because junit uses reflection and the default SecurityManager prevents that.
+	 */
+	private void executeWithSecurityManager(Runnable runnable) {
+		Policy.getPolicy().refresh();
```

Comment:
Does something like that needs to be called after the SystemProperty java.security.policy is reverted after the test?

Reply:
According to the [official Oracle Java SE documentation](https://docs.oracle.com/javase/8/docs/api/index.html?java/security/Policy.html)
> The `refresh` method causes the policy object to refresh/reload its data. This operation is implementation-dependent. For example, if the policy object stores its data in configuration files, calling `refresh` will cause it to re-read the configuration policy files. If a refresh operation is not supported, this method does nothing. Note that refreshed policy may not have an effect on classes in a particular ProtectionDomain. This is dependent on the Policy provider's implementation of the `implies` method and its PermissionCollection caching strategy.

Reply:
If you want to test if resetting the policy works as expected, you should probably add a new test for that and add `@Order` to your suite. [Here is a blog post from Baeldung on that topic]()https://www.baeldung.com/junit-5-test-order.